validator/client/execution_payload.go

```diff
@@ -0,0 +1,53 @@
+package client
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/signing"
+	fieldparams "github.com/prysmaticlabs/prysm/v5/config/fieldparams"
+	"github.com/prysmaticlabs/prysm/v5/config/params"
+	enginev1 "github.com/prysmaticlabs/prysm/v5/proto/engine/v1"
+	validatorpb "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1/validator-client"
+	"github.com/prysmaticlabs/prysm/v5/time/slots"
+)
+
+func (v *validator) signExecutionPayloadEnvelope(ctx context.Context, p *enginev1.ExecutionPayloadEnvelope, pubKey [fieldparams.BLSPubkeyLength]byte) ([]byte, error) {
+	// Get domain data
+	currentSlot := slots.CurrentSlot(v.genesisTime)
+	epoch := slots.ToEpoch(currentSlot)
+	domain, err := v.domainData(ctx, epoch, params.BeaconConfig().DomainBeaconBuilder[:])
+	if err != nil {
+		return nil, errors.Wrap(err, domainDataErr)
+	}
+	if domain == nil {
+		return nil, errors.New(domainDataErr)
+	}
+
+	// Compute signing root
+	signingRoot, err := signing.ComputeSigningRoot(p, domain.SignatureDomain)
+	if err != nil {
+		return nil, errors.Wrap(err, signingRootErr)
+	}
+
+	// Create signature request
+	signReq := &validatorpb.SignRequest{
+		PublicKey:       pubKey[:],
+		SigningRoot:     signingRoot[:],
+		SignatureDomain: domain.SignatureDomain,
+		Object:          &validatorpb.SignRequest_ExecutionPayloadEnvelope{ExecutionPayloadEnvelope: p},
+		SigningSlot:     currentSlot,
+	}
+
+	// Sign the payload attestation data
```

Comment:
```suggestion
	// Sign the execution payload envelope.
```
`execution_payload_header.go` also has a wrong comment.

`payload_attestation.go`, `execution_payload.go`, and `execution_payload_header.go` have identical methods except `SignRequest.SignatureDomain` and `SignRequest.Object`. Doesn't look like a problem at the moment, but can't deny that it doesn't smell good.